nicegui/elements/sub_pages.js

```diff
@@ -0,0 +1,67 @@
+function getPathPrefix() {
+  return window.path_prefix || "";
+}
+
+function getCurrentPath() {
+  return window.location.pathname + window.location.search + window.location.hash;
+}
+
+function stripPathPrefix(path) {
+  const prefix = getPathPrefix();
+  return prefix && path.startsWith(prefix) ? path.substring(prefix.length) : path;
+}
+
+function getCleanCurrentPath() {
+  return stripPathPrefix(getCurrentPath());
+}
+
+function buildFullPath(cleanPath) {
+  return getPathPrefix() + cleanPath;
+}
+
+function handleStateEvent(event) {
+  const cleanPath = event.state?.page || getCleanCurrentPath();
+  emitEvent("sub_pages_open", cleanPath);
+}
+
+function handleFragmentNavigation(href, targetUrl) {
+  const fragmentName = targetUrl.hash.substring(1);
+  const target = document.getElementById(fragmentName) || document.querySelector(`a[name="${fragmentName}"]`);
+  if (!target) return false;
+
+  target.scrollIntoView({ behavior: "smooth" });
```

Comment:
While our NiceGUI website uses smooth scrolling, other apps might prefer a different behavior. Maybe we should make it configurable.

Reply:
Sure. But we could do this in later updates. Or do you think it will likely break the existing API?